karapace/backup/topic_configurations.py

```diff
@@ -0,0 +1,73 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from enum import Enum
+from kafka import KafkaAdminClient
+from kafka.admin import ConfigResource, ConfigResourceType
+from kafka.errors import for_code
+from kafka.protocol.admin import DescribeConfigsRequest
+from typing import Dict, Iterable, Optional
+
+
+class ConfigSource(int, Enum):
+    UNKNOWN = 0
+    TOPIC_CONFIG = 1
+    DYNAMIC_BROKER_CONFIG = 2
+    DYNAMIC_DEFAULT_BROKER_CONFIG = 3
+    STATIC_BROKER_CONFIG = 4
+    DEFAULT_CONFIG = 5
+    DYNAMIC_BROKER_LOGGER_CONFIG = 6
```

Comment:
question: Do we want to record these sources in the backups?

Reply:
I don't believe it's really necessary, because all the configs that are backed up will be restored anyway. I would avoid polluting the metadata record with too much information.

Reply:
✅